stories/setup.stories.js

```diff
@@ -27,94 +27,130 @@ import { storiesOf } from '@storybook/react';
 import SetupUsingGCP from '../assets/js/components/legacy-setup/SetupUsingGCP';
 import SetupUsingProxy from '../assets/js/components/setup/SetupUsingProxy';
 import { CORE_USER, DISCONNECTED_REASON_CONNECTED_URL_MISMATCH } from '../assets/js/googlesitekit/datastore/user/constants';
-import { provideUserAuthentication, WithTestRegistry } from '../tests/js/utils';
+import { createTestRegistry, provideSiteConnection, provideUserAuthentication, WithTestRegistry } from '../tests/js/utils';
+
+const withRegistry = ( Story ) => {
+	const registry = createTestRegistry();
+	registry.dispatch( CORE_USER ).receiveGetTracking( { enabled: false } );
+
+	return (
+		<Story registry={ registry } />
+	);
+};
 
 storiesOf( 'Setup / Using GCP', module )
-	.add( 'Step one', () => {
+	.add( 'Step one', ( args, { registry } ) => {
 		global._googlesitekitLegacyData.setup.isSiteKitConnected = false;
 		global._googlesitekitLegacyData.setup.isAuthenticated = false;
 		global._googlesitekitLegacyData.setup.isVerified = false;
 		global._googlesitekitLegacyData.setup.hasSearchConsoleProperty = false;
 		global._googlesitekitLegacyData.permissions.canSetup = true;
 
-		const setupRegistry = ( { dispatch } ) => {
-			dispatch( CORE_USER ).receiveGetAuthentication( {
-				authenticated: false,
-				requiredScopes: [],
-				grantedScopes: [],
-			} );
-
-			dispatch( CORE_USER ).receiveGetTracking( {
-				enabled: false,
-			} );
-		};
+		registry.dispatch( CORE_USER ).receiveGetAuthentication( {
```

Comment:
Should we use `provideUserAuthentication` instead? If not, we should probably annotate why here.